### PR TITLE
walletrpc: add basic coin selection RPCs

### PR DESCRIPTION
**Pull request by wpaulino**
This PR adds the minimum set of RPCs required for a higher-level application to perform coin selection backed by the embedded btcwallet instance.

These RPCs include:
* `ListUnspent`: a copy of the existing `ListUnspent` RPC in `LightningClient`, which has been marked as deprecated in favor of using the `walletrpc` one.
* `LeaseOutput`: lease an output for a predetermined amount of time bound to a unique ID to ensure no other applications can lease the same output.
* `ReleaseOutput`: if the lease for an output is no longer needed, it can be returned to the set of spendable UTXOs manually through this RPC, or it will be done automatically by the wallet once the lease expires.

~~Depends on https://github.com/btcsuite/btcwallet/pull/700.~~
Partially addresses https://github.com/lightningnetwork/lnd/issues/2991.

### CODE REVIEW

File: lnwallet/interface.go [hunk 1]
```diff
@@ -221,6 +223,22 @@ type WalletController interface {
 	// eligible for coin selection.
 	UnlockOutpoint(o wire.OutPoint)
 
+	// LeaseOutput locks an output to the given ID, preventing it from being
```
Comment (Roasbeef) [inline lnwallet/interface.go:226]: Should really start to split up this interface as we've done for the `keychain` related interfaces lately, but that's outside the scope fo this PR.

File: lnrpc/walletrpc/walletkit.proto [hunk 2]
```diff
@@ -108,6 +124,35 @@ service WalletKit {
...
 
+message LeaseOutputRequest {
+    /*
+    An ID of 32 random bytes that must be unique for each distinct application
+    using this RPC which will be used to bound the output lease to.
+    */
+    bytes id = 1;
+
+    // The identifying outpoint of the output being leased.
+    lnrpc.OutPoint outpoint = 2;
```
Comment (Roasbeef) [inline lnrpc/walletrpc/walletkit.proto:154]: We use "outpoint" here, but the calls all use "output". Thinking we should make it consistent here and use "outpoint" everywhere on the RPC level.

Comment (wpaulino) [reply lnrpc/walletrpc/walletkit.proto:154]: Well you really are leasing the output itself, with outpoint being how you refer to it.

File: lnrpc/walletrpc/walletkit_server.go [hunk 3]
```diff
@@ -205,6 +205,38 @@ func (w *WalletKit) RegisterWithRootServer(grpcServer *grpc.Server) error {
...
+	// the set of UTXOs that match our query.
+	utxos, err := w.cfg.Wallet.ListUnspentWitness(minConfs, maxConfs)
+	if err != nil {
+		return nil, err
+	}
+
+	rpcUtxos, err := lnrpc.MarshalUtxos(utxos, w.cfg.ChainParams)
+	if err != nil {
+		return nil, err
+	}
```
Comment (cfromknecht) [inline lnrpc/walletrpc/walletkit_server.go:234]: nit: add new line

**Comment by Roasbeef** [comment]
This should also fix #3270 if updates to the latest btcwallet commit.

**Comment by cfromknecht** [comment]
@wpaulino needs rebase

**Comment by wpaulino** [comment]
Forgot to include the macaroon permissions for the new RPCs, should be good now.

**Pull request merged**